File: 414-third-maximum-number/third-maximum-number.java
class Solution {
    public int thirdMax(int[] nums) {
        double max1 = Double.NEGATIVE_INFINITY;
        double max2 = Double.NEGATIVE_INFINITY;
        double max3 = Double.NEGATIVE_INFINITY;
        for (int num : nums){
            if (num>max1){
                max3=max2;
                max2=max1;
                max1=num;
            }else if(num<max1 && num>max2){
                max3=max2;
                max2=num;
            }else if(num<max2 && num>max3){
                max3=num;
            }

        }
        if (max3 == Double.NEGATIVE_INFINITY) {
            return (int) max1;
        } else {
            return (int) max3;
        }
        
    }
}

File: 414-third-maximum-number/third-maximum-number.py
from typing import List

class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        maxx1 = -2147483649
        maxx2 = -2147483649
        maxx3 = -2147483649
        
        for i in nums:
            if i > maxx1:
                maxx3 = maxx2
                maxx2 = maxx1
                maxx1 = i
            elif i > maxx2 and i < maxx1:
                maxx3 = maxx2
                maxx2 = i
            elif i > maxx3 and i < maxx2:
                maxx3 = i
        
        if maxx3 == -2147483649:
            return maxx1
        else:
            return maxx3
